Composer/packages/extensions/obiformeditor/src/Form/fields/CustomObjectArrayField.tsx

```diff
@@ -0,0 +1,231 @@
+        <EditableField
+          onChange={handleEdit(keyArray[0])}
+          value={item[keyArray[0]]}
+          placeholder={itemProperties[keyArray[0]].description}
+          styles={{
+            root: { margin: '7px 0 7px 0' },
+          }}
+          onBlur={handleBlur}
+        />
+      </div>
+      <div css={customObjectArrayItemValue}>
+        <EditableField
+          onChange={handleEdit(keyArray[1])}
+          value={item[keyArray[1]]}
+          placeholder={itemProperties[keyArray[1]].description}
+          styles={{
+            root: { margin: '7px 0 7px 0' },
+          }}
+          options={{ transparentBorder: true }}
+          onBlur={handleBlur}
+        />
+      </div>
+      <div>
+        <IconButton
+          menuProps={{ items: contextItems }}
+          menuIconProps={{ iconName: 'MoreVertical' }}
+          ariaLabel={formatMessage('Item Actions')}
+          styles={{ menuIcon: { color: NeutralColors.black, fontSize: FontSizes.size16 } }}
+        />
+      </div>
+    </div>
+  );
+};
+
+export const CustomObjectArrayField: React.FC<FieldProps> = props => {
+  const { onChange, formData = [], id, label, schema = {} } = props;
+  const description = schema.description;
+  const itemProperties = (schema.items as JSONSchema6).properties as { [k: string]: JSONSchema6 };
+  const keyArray = Object.keys(itemProperties) || [];
+  const [newItem, setNewItem] = useState<{ [key: string]: any } | null>(null);
+  const [errorMsg, setErrorMsg] = useState<string>('');
+
+  const handleReorder = (aIdx: number, bIdx: number) => {
+    onChange(swap(formData, aIdx, bIdx));
+  };
+
+  const handleDelete = (idx: number) => {
+    onChange(remove(formData, idx));
+  };
+
+  const handleEdit = (idx, val) => {
+    const items = [...(formData || [])];
+    items[idx] = val;
+    onChange(items);
+  };
+
+  const handleNewItemEdit = (field: string) => (_e: any, data?: string) => {
+    setNewItem({ ...newItem, [field]: data });
+  };
+
+  const handleKeyDown = (e: React.KeyboardEvent) => {
+    if (e.key.toLowerCase() === 'enter') {
+      e.preventDefault();
+
+      if (newItem) {
+        if (newItem[keyArray[0]]) {
+          onChange([...formData, newItem]);
+          setNewItem(null);
+          setErrorMsg('');
+        } else {
+          setErrorMsg(formatMessage('value required'));
+        }
+      }
+    }
+  };
+
+  return (
+    <React.Fragment>
+      <WidgetLabel label={label} description={description} id={id} />
+      <div css={[customObjectArrayItemContainer('flex-start'), customObjectArrayItemLabel]}>
+        <div css={[customObjectArrayItemValue, customObjectArrayItemValueLabel]}>
+          {itemProperties[keyArray[0]].title}
+        </div>
+        <div css={[customObjectArrayItemValue, customObjectArrayItemValueLabel]}>
+          {itemProperties[keyArray[1]].title}
+        </div>
+      </div>
+      <div css={customObjectArrayField}>
+        {formData &&
```

Comment:
`formData` had already coalesced to `[]`.
```suggestion
        {
```